synfig-studio/src/gui/app.cpp

```diff
@@ -4390,9 +4397,9 @@ void
 studio::App::process_all_events(long unsigned int us)
 {
 	Glib::usleep(us);
```

Comment:
> Glib::usleep(us);

As I understand the current logic, `usleep` function was supposed to stop the execution of the current thread (for example, the rendering thread) and allow the main GUI thread to handle events. But that's not how it works, so it's not needed here.

> gtk_main_iteration_do(false);

There is no difference in true/false here, because the function blocks only when there are no pending events. But we've already checked the pending events, so I prefer `true` here, just to show that we know what we're doing.

Second while loop is also not needed here.

I prefer `gtk_main_iteration_do(false);`

Reply:
I think the second while loop is there because someone relied on this statement you made:
> As I understand the current logic, usleep function was supposed to stop the execution of the current thread (for example, the rendering thread) and allow the main GUI thread to handle events.